yarn-project/pxe/src/synchronizer/synchronizer.ts

```diff
@@ -286,7 +288,12 @@ export class Synchronizer {
   public getSyncStatus() {
     return {
       blocks: this.synchedToBlock,
-      notes: Object.fromEntries(this.noteProcessors.map(n => [n.publicKey.toString(), n.status.syncedToBlock])),
+      notes: Object.fromEntries(
+        [...this.noteProcessors, ...this.noteProcessorsToCatchUp].map(n => [
+          n.publicKey.toString(),
+          n.status.syncedToBlock,
+        ]),
+      ),
```

Comment:
This change, together with how `SentTx.wait` works[1], will mean that by default all sent transaction will wait for all notes processors registered on the PXE to catch up with the latest block.


https://github.com/AztecProtocol/aztec-packages/blob/5d93a470122aaddfcbd1e59e54568357df871098/yarn-project/aztec.js/src/contract/sent_tx.ts#L106-L109

Reply:
I'm considering moving this behaviour (waiting for notes processors to catch up) just to `DeployAccountTxReceipt` (which is a specialised version of `SentTx` for accounts) and have its `wait` method wait only for its own note processor to finish instead of all.

I think reporting the status of catching up note processors is good but I worry the way it's coupled to arbitrary transactions might cause performance problems.

Reply:
I wouldn't make this change yet. I tend to agree that I think we need a different model of reporting/waiting for sync status but I'd like to take a more thorough look at it once we have some kind of note discovery in place.